core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -937,7 +937,7 @@ private[spark] object Utils extends Logging {
       // In non-Yarn mode (or for the driver in yarn-client mode), we cannot trust the user
       // configuration to point to a secure directory. So create a subdirectory with restricted
       // permissions under each listed directory.
-      conf.get("spark.local.dir", System.getProperty("java.io.tmpdir")).split(",")
+      randomizeInPlace(conf.get("spark.local.dir", System.getProperty("java.io.tmpdir")).split(","))
```

Comment:
if we are calling it in all cases then we should just get the dirs and then call it at the end.

if val dirs = if (isRunningInYarnContainer(conf) {....}
// return:
randomizeInPlace(dirs)

Reply:
+1

Reply:
+1